File: archive-romhandbook-ui/lib/types/Headwear.ts
export type ApiResponse<T> = {

    success: boolean

    data: T

    meta: unknown

}

export type PaginatedApiResponse<T> = {

    success: boolean

    data: T[]

    meta: {
        page: number
        limit: number
        total: number
        has_next: boolean
    }

}

export type Headwear = {

    id: string

    detail_url: string

    image: string | null

    name: string

    type: string | null

    description: string | null

    quality: string | null

    effect_text: string | null

    unlock_text: string | null

    deposit_stats: string | null

    unlock_stats: string | null

    jobs: string | null

    availability_date: string | null

}

export type HeadwearFormula = {

    id: number

    headwear_id: string

    formula_index: number | null

    formula_json: string | null

}

export type HeadwearDetail = Headwear & {

    formula_id: string | null

    raw_html: string | null

    formulas: HeadwearFormula[]

}

File: backend-api/internal/models/headwear.go
package models

type Headwear struct {
	ID string `json:"id"`

	DetailURL string `json:"detail_url"`

	Image *string `json:"image"`

	Name string `json:"name"`

	Type *string `json:"type"`

	Description *string `json:"description"`

	Quality *string `json:"quality"`

	EffectText *string `json:"effect_text"`

	UnlockText *string `json:"unlock_text"`

	DepositStats *string `json:"deposit_stats"`

	UnlockStats *string `json:"unlock_stats"`

	Jobs *string `json:"jobs"`

	AvailabilityDate *string `json:"availability_date"`
}

type HeadwearDetail struct {
	ID string `json:"id"`

	DetailURL string `json:"detail_url"`

	Image *string `json:"image"`

	Name string `json:"name"`

	Type *string `json:"type"`

	Description *string `json:"description"`

	Quality *string `json:"quality"`

	EffectText *string `json:"effect_text"`

	UnlockText *string `json:"unlock_text"`

	DepositStats *string `json:"deposit_stats"`

	UnlockStats *string `json:"unlock_stats"`

	Jobs *string `json:"jobs"`

	FormulaID *string `json:"formula_id"`

	AvailabilityDate *string `json:"availability_date"`

	RawHTML *string `json:"raw_html"`

	Formulas []HeadwearFormula `json:"formulas"`
}

type HeadwearFormula struct {
	ID int `json:"id"`

	HeadwearID string `json:"headwear_id"`

	FormulaIndex *int `json:"formula_index"`

	FormulaJSON *string `json:"formula_json"`
}
